charmrepo/charmstore_test.go

```diff
@@ -94,18 +94,55 @@ type charmStoreRepoSuite struct {
 
 var _ = gc.Suite(&charmStoreRepoSuite{})
 
+// checkCharmDownloads checks that the charm represented by the given URL has
+// been downloaded the expected number of times.
+func (s *charmStoreRepoSuite) checkCharmDownloads(c *gc.C, url *charm.URL, expect int) {
+	client := csclient.New(csclient.Params{
+		URL: s.srv.URL(),
+	})
+
+	key := []string{params.StatsArchiveDownload, url.Series, url.Name, url.User, strconv.Itoa(url.Revision)}
+	path := "/stats/counter/" + strings.Join(key, ":")
+	var count int
+
+	getDownloads := func() int {
+		var result []params.Statistic
+		err := client.Get(path, &result)
+		c.Assert(err, jc.ErrorIsNil)
+		return int(result[0].Count)
+	}
+
+	for retry := 0; retry < 10; retry++ {
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Is waiting the best way to do this? Should we not instead wait on a trigger of some sort?